components/Router.js

```diff
@@ -0,0 +1,54 @@
+import { render } from '@asyncapi/generator-react-sdk';
+import { pascalCase } from './common';
+
+let addHandlerFunction = (queue, operation) => `
+  r.AddNoPublisherHandler(
+    "${operation}",     // handler name, must be unique
+    "${queue}",         // topic from which we will read events
+    s,
+    handlers.${operation}, 
+  )
+`
+
+function AMQPRouterRules({ channels }) {
+  return Object.entries(channels)
+  .map(([channelName, channel]) => {
+    if ( channel.hasPublish()) {
+      let operation = pascalCase(channel.publish().id())
+      let queue = channel.bindings().amqp.queue.name
+      return  addHandlerFunction(queue, operation);
+    }
+    return "";
+  });
+}
+
+export function Router({moduleName, channels, subscriberFlags}) {
+
+  let amqpRules = ""
+  if (subscriberFlags.hasAMQPSub) {
+    amqpRules = `
+//ConfigureAMQPSubscriptionHandlers configures the router with the subscription handler    
```

Comment:
There is an small but important fix to do here. **All** documented types in go should have an space at the beginning of the comment. A minor also is adding a final period. All is by convention. 
```suggestion
// ConfigureAMQPSubscriptionHandlers configures the router with the subscription handler.    
```

Same applies to all comments your wrote.